src/cpu/dyn_cache.h

```diff
@@ -62,38 +72,32 @@ class CacheBlock {
 		toblock->link[index].from = this; // remember who links me
 	}
 
-	struct {
+	struct Page {
 		uint16_t start = 0;
 		uint16_t end   = 0; // where in the page is the original code
 
 		CodePageHandler* handler = {}; // page containing this code
 	} page = {};
 
-	struct {
-		// TODO field start used to be a normal pointer, but upstream
-		// changed it to const pointer in r4424 (perhaps by mistake or
-		// as WIP change). Once transition to W^X will be done, decide
-		// if this should be const pointer or not and remove this comment.
-		//
-		// uint8_t *start; // where in the cache are we
+	struct Cache {
 		const uint8_t* start = {}; // where in the cache are we
 
 		Bitu size        = 0;
 		CacheBlock* next = {};
 		// writemap masking maskpointer/start/length to allow holes in
 		// the writemap
-		std::unique_ptr<uint8_t[]> wmapmask = {};
+		uint8_t* wmapmask = {};
```

Comment:
When taken at face value - std::vector is the right object for this because it even copies the previous content, and would let us eliminate the annoying 'len' variable too! 

The problem is, the dynrec code relies on getting the [`offsetof`](https://en.cppreference.com/w/cpp/types/offsetof) the Page::start members, and as of C++17, all dynamic containers like `std::vector` and even smart-pointers produce [a non-standard-layout](https://quuxplusone.github.io/blog/2022/03/04/non-standard-layout-guarantees/), however `offsetof` relies on having a standard layout.

Long ago I tried using vector and couldn't - so fell back to the smart pointer which both GCC and older Clang's allowed. But it looks like the newer Clang's (which this PR moves toward) now catch this for the smart pointers too :-)

If you can try (if you have a new-ish), you should see a handful of these from Clang:

"warning: offset of on non-standard-layout type 'CacheBlock'"

Build with (if you have Clang 14):  

```shell
meson setup -Dbuildtype=debug --native-file=.github/meson/native-clang-14.ini build/clang-debug
meson compile -C build/clang-debug
```